### PR TITLE
stat var hierarchy bugs

### PR DESCRIPTION
**Pull request by chejennifer**
fixed bugs:
- when section is expanded and user selects a search result within that section, result won't show up.
- when user clears search result, the expanded section (from the selected search result) is rendered and then closed

screen-recording (16)

### CODE REVIEW

File: static/js/browser2/statvar_group_node.tsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ interface StatVarGroupNodePropType {
 }
 
 interface StatVarGroupNodeStateType {
-  isRendered: boolean;
+  toggledOpen: boolean;
```
Comment (beets) [inline static/js/browser2/statvar_group_node.tsx:56]: add a comment here? specifically how this relates to the prop `open`. maybe the prop should be renamed to something like `startsOpened`

**Review by beets — APPROVED** [review]
thanks for the bug fix

File: static/js/browser2/statvar_group_node.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/browser2/statvar_group_node.tsx:59]: done

File: static/js/browser2/statvar_hierarchy.tsx [hunk 2]
```diff
@@ -86,7 +89,16 @@ export class StatVarHierarchy extends React.Component<
...
 
+  componentDidUpdate(): void {
+    if (this.state.searchSelectionCleared) {
+      this.setState({ searchSelectionCleared: false });
+    }
+  }
+
   render(): JSX.Element {
+    if (this.state.searchSelectionCleared) {
+      return null;
```
Comment (shifucun) [inline static/js/browser2/statvar_hierarchy.tsx:100]: why return null here? add some comments?

**Review by shifucun — COMMENTED** [review]
Tricky states management! Can you also help check the hierarchy in /dev works as intended.

File: static/js/browser2/statvar_hierarchy.tsx (hunk 2 above)
Comment (chejennifer) [reply static/js/browser2/statvar_hierarchy.tsx:103]: done.

**Comment by chejennifer** [comment]
> Tricky states management! Can you also help check the hierarchy in /dev works as intended.

Yeup, I have checked /dev

**Review by shifucun — APPROVED** [review]

**Comment by shifucun** [comment]
> > Tricky states management! Can you also help check the hierarchy in /dev works as intended.
> 
> Yeup, I have checked /dev

Thanks!

**Pull request merged**